dune

```diff
@@ -1,19 +1,18 @@
 (env
- (development
+ (dev
   (flags (:standard
           -strict-formats  ;; Disallows legacy formats.
           -strict-sequence ;; Enforces the lhs of a sequence to have type `unit'.
           -safe-string     ;; Immutable strings.
           -annot           ;; Dumps information per (file) module about types, bindings, tail-calls, etc. Used by some tools.
-          -w Ae-44-45-60   ;; Ignores warnings 44, 45, and 60.
+          -w A-4-44-45-60  ;; Ignores warnings 4, 44, 45, and 60.
           -g               ;; Adds debugging information to the resulting executable / library.
          )))
  (release
   (flags (:standard
           ;; The following flags are the same as for the "development" profile.
           -strict-formats -strict-sequence -safe-string -annot
-          ;; The following flags are unique to the "release" profile.
-          -w @a-4-44-45-60 ;; Treats all warnings as errors, except that 4, 44, 45, and 60 are ignored altogether.
+          -w A-4-44-45-60 ;; Ignore warnings 4, 44, 45, and 60.
```

Comment:
I think we should still error on warnings when using the `release` profile. However, we can allow a bit more slack. I think deprecation warnings have the number `3`, thus if you add that to the list those warnings should be ignored.

Reply:
Part of the issue here is that some other warnings have become more aggressive in OCaml 4.08.0. The following now warns, as we're popping from a stack of functions, and so it considers the function partially applied (due to warning no. 5).

https://github.com/links-lang/links/blob/145dc45c389e8790779c31676a21fe48d91823ad/core/lexer.mll#L149

It should be OK to fix it, but I feel it's safer to leave it off to avoid similar breaking changes in the future.

Reply:
Rewriting it as `ignore (Stack.pop lexers)` may eliminate the warning.

Reply:
Sadly [it turns out it isn't](https://github.com/ocaml/ocaml/blob/658233568930c9b1c291f78d8ac2177e67aa9981/typing/typecore.ml#L4149)! Resorting to `[@warning "-5"]` works though.